portfolio/src/main/java/com/google/sps/data/CuisineDatastore.java

```diff
@@ -0,0 +1,88 @@
+package com.google.sps.data;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.EnumMap;
+import com.google.appengine.api.datastore.*;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.AvailableCuisines;
+
+/** Class that handles writing/reading cuisine data from the datastore. */
+public final class CuisineDatastore {
+  
+  public static final String CUISINE_ENTITY = "CuisineVotes";
+  public static final String CUISINE_PARAMETER = "cuisine";
+  public static final String USERID_PARAMETER = "userId";
+
+  private CuisineDatastore() {}
+
+    /** Method that retreives cuisine entities from the datastore and formats it. */
+    public static Map<AvailableCuisines, Integer> fetchCuisineVotes() {
+      Query query = new Query(CUISINE_ENTITY);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+      
+      EnumMap<AvailableCuisines, Integer> cuisineVotes = new EnumMap<>(AvailableCuisines.class);
+
+      for (AvailableCuisines cuisineId : AvailableCuisines.values()) {
+        cuisineVotes.put(cuisineId, 0);
+      }
+
+      for (Entity entity : results.asIterable()) {
+        String cuisine = (String) entity.getProperty(CUISINE_PARAMETER); //store the name of the enum (capitalized)
+        AvailableCuisines cuisineId = AvailableCuisines.getFromId(cuisine);
```

Comment:
should you try/catch for IllegalArgs here and ignore the entry. This can definitely happen if you remove entries from your cuisine enum later after launching the project.